contrib/mesos/pkg/executor/executor.go

```diff
@@ -212,10 +199,15 @@ func (k *KubernetesExecutor) Registered(driver bindings.ExecutorDriver,
 	}
 
 	if executorInfo != nil && executorInfo.Data != nil {
-		k.staticPodsConfig = executorInfo.Data
+		k.initializeStaticPodsSource(executorInfo.Data)
 	}
 
-	k.initialRegistration.Do(k.onInitialRegistration)
+	k.updateChan <- kubelet.PodUpdate{
```

Comment:
I have added the missing locking. The other channel writes are fine already. But one has to do quite some control flow analysis to check that. This should be much more obvious.